control/control.go

```diff
@@ -911,8 +1003,43 @@ func (p *pluginControl) CollectMetrics(metricTypes []core.Metric, deadline time.
 
 	// For each available plugin call available plugin using RPC client and wait for response (goroutines)
 	for pluginKey, pmt := range pluginToMetricMap {
+
+		// if plugin does not have a pool, try to make new subscription
+		if !p.pluginRunner.AvailablePlugins().hasPool(pluginKey) {
+			// this happens only when the queried metric in task manifest matches with metrics
```

Comment:
@jcooklin, yes, it stricte related with discussion about pool subscription that we had and this should be removed from CollectMetrics. But the question is if we would like to:
- create new issue about mechanism which will validate metrics type subscriptions, subscribe collector deps and allow to create new pool not - only in the phase of creating task, but also during its running (for case when metrics exposed by already loaded plugin match with requested metrics namespaces)
- suspend this PR until all of that mentioned above will be implemented